activestorage/test/controllers/direct_uploads_controller_test.rb

```diff
@@ -0,0 +1,121 @@
+require "test_helper"
+require "database/setup"
+
+if SERVICE_CONFIGURATIONS[:s3]
+  class ActiveStorage::S3DirectUploadsControllerTest < ActionDispatch::IntegrationTest
+    setup do
+      @old_service = ActiveStorage::Blob.service
+      ActiveStorage::Blob.service = ActiveStorage::Service.configure(:s3, SERVICE_CONFIGURATIONS)
+    end
+
+    teardown do
+      ActiveStorage::Blob.service = @old_service
+    end
+
+    test "creating new direct upload" do
+      checksum = Digest::MD5.base64digest("Hello")
+
+      post rails_direct_uploads_url, params: { blob: {
+        filename: "hello.txt", byte_size: 6, checksum: checksum, content_type: "text/plain" } }
+
+      response.parsed_body.tap do |details|
+        assert_equal ActiveStorage::Blob.find(details["id"]), ActiveStorage::Blob.find_signed(details["signed_id"])
+        assert_equal "hello.txt", details["filename"]
+        assert_equal 6, details["byte_size"]
+        assert_equal checksum, details["checksum"]
+        assert_equal "text/plain", details["content_type"]
+        assert_match /#{SERVICE_CONFIGURATIONS[:s3][:bucket]}\.s3.(\S+)?amazonaws\.com/, details["direct_upload"]["url"]
```

Comment:
This test is currently failing if the the name of the AWS bucket includes a `.`.

For instance, if the bucket is called `rails.test` then the direct upload URL returned by AWS can be:

"https://s3.amazonaws.com/rails.test/xucyA48o9DbySmKBP…"

rather than

"https://rails.test.s3.amazonaws.com/xucyA48o9DbySmKBP…"